Submodule DanaKit contains modified content
diff --git a/DanaKit/DanaKit/PumpManager/DanaKitPumpManager.swift b/DanaKit/DanaKit/PumpManager/DanaKitPumpManager.swift
index d795d59..88ca56b 100644
--- a/DanaKit/DanaKit/PumpManager/DanaKitPumpManager.swift
+++ b/DanaKit/DanaKit/PumpManager/DanaKitPumpManager.swift
@@ -556,18 +556,6 @@ extension DanaKitPumpManager: PumpManager {
                         self.state.bolusState = .inProgress
                         self.notifyStateDidChange()
                         
-                        DispatchQueue.main.async {
-                            let dose = self.doseEntry?.toDoseEntry(isMutable: true)
-                            
-                            if let dose = dose {
-                                self.pumpDelegate.notify { (delegate) in
-                                    delegate?.pumpManager(self, hasNewPumpEvents: [NewPumpEvent.bolus(dose: dose, units: dose.programmedUnits)], lastReconciliation: Date.now, replacePendingEvents: false, completion: { _ in })
-                                }
-                                
-                                self.notifyStateDidChange()
-                            }
-                        }
-                        
                         completion(nil)
                     } catch {
                         self.state.bolusState = .noBolus
